### PR TITLE
[astro] Incorporated new QuantityType (Units of Measurement)

### LINKED ISSUE #5775: [Astro] Use UoM for moon distances and sun radiation
The astro binding has kilometer channels and miles channels for moon distance, perigee and apogee.
Could you please convert this to only one channel each with Number:Length type?

Also please change the sun radiation channels to Number:Intensity type.

### PR DESCRIPTION
**Pull request by cweitkamp**
- Incorporated new QuantityType (Units of Measurement)

Please add the APIBreaking label. The `kilometer` and `miles` channels have been replaced by a new  `distance` channel.

Closes #5775

Signed-off-by: Christoph Weitkamp <github@christophweitkamp.de>

### CODE REVIEW

File: extensions/binding/org.eclipse.smarthome.binding.astro/ESH-INF/thing/channels.xml [hunk 1]
```diff
@@ -336,31 +337,31 @@
...
+		<item-type>Number:Angle</item-type>
 		<label>Moon age</label>
-		<description>The age of the moon in degree</description>
-		<state readOnly="true" pattern="%.2f °" />
+		<description>The age of the moon</description>
+		<state readOnly="true" pattern="%.2f %unit%" />
 	</channel-type>
 
 	<channel-type id="agePercent">
 		<item-type>Number</item-type>
```
Comment (NorbertHD) [inline extensions/binding/org.eclipse.smarthome.binding.astro/ESH-INF/thing/channels.xml]: I'm not an expert in UoM, but shouldn't this be Number:Dimensionless?

Comment (cweitkamp) [reply extensions/binding/org.eclipse.smarthome.binding.astro/ESH-INF/thing/channels.xml]: Absolutely. Thanks for the hint. I will change it.

File: extensions/binding/org.eclipse.smarthome.binding.astro/src/main/java/org/eclipse/smarthome/binding/astro/internal/model/Range.java [hunk 2]
```diff
@@ -52,15 +57,15 @@ public Calendar getEnd() {
...
-            return -1;
+            return null;
         }
         if (start.after(end)) {
-            return 0;
+            return new QuantityType<Time>(0, SmartHomeUnits.MINUTE);
         }
         long diff = end.getTimeInMillis() - start.getTimeInMillis();
-        return diff / 60000;
+        return new QuantityType<Time>(diff / 60000, SmartHomeUnits.MINUTE);
```
Comment (wborn) [inline extensions/binding/org.eclipse.smarthome.binding.astro/src/main/java/org/eclipse/smarthome/binding/astro/internal/model/Range.java]: You could get rid of the magic number 60000 by using the QuantityType itself for doing the ms to minutes conversion:

```java
long diff = end.getTimeInMillis() - start.getTimeInMillis();
QuantityType millis = new QuantityType<>(diff, MetricPrefix.MILLI(SmartHomeUnits.SECOND));
return millis.toUnit(SmartHomeUnits.MINUTE);
```

**Comment by cweitkamp** [comment]
FTR: I have migrated all the unit tests from Groovy to pure Java for this binding. Shall I append those changes to this PR or submit an follow-up PR?

**Comment by sjka** [comment]
That's awesome! 🏅 
A separate PR would be the cleaner option, I think.

**Comment by sjka** [comment]
It looks good from my point of view - @gerrieg as the original author, would you want to have a look too?

**Comment by sjka** [comment]
Thanks!

**Comment by kaikreuzer** [comment]
> Please add the APIBreaking label. The kilometer and miles channels have been replaced by a new distance channel.

@cweitkamp May I ask you to add this info also to https://github.com/openhab/openhab-distro/blob/master/distributions/openhab/src/main/resources/bin/update.lst, so that people are warned when updating their installations?

**Comment by cweitkamp** [comment]
@kaikreuzer Done (PR openhab/openhab-distro#748)

**Pull request merged**